walk-in-interview/src/main/webapp/script.js

```diff
@@ -122,98 +109,205 @@ function renderSelectOptions(id) {
-      });
+function displayJobListings(jobPageData) {
   const jobListingsElement = document.getElementById('job-listings');
+  const jobShowing = document.getElementById('job-listings-showing');
 
-  /** reset the list so we don't render the same jobs twice */
+  /* reset the list so we don't render the same jobs twice */
   jobListingsElement.innerHTML = '';
+  jobShowing.innerHTML = '';
+
+  if (jobPageData === undefined ||
+    !jobPageData.hasOwnProperty('jobList') ||
+    jobPageData['jobList'].length === 0) {
+    setErrorMessage(/* msg= */ STRINGS['no-jobs-error-message'],
+        /* includesDefaultMsg= */ false);
+    return;
+  }
+
+  const jobListings = jobPageData['jobList'];
+
   const jobListingTemplate = document.getElementById('job-listing-template');
 
   jobListings.forEach((job) => {
     const jobListing =
-      jobListingTemplate.cloneNode( /** and child elements */ true);
+      jobListingTemplate.cloneNode( /* and child elements */ true);
     jobListing.setAttribute('id', job['jobId']);
 
     const jobTitle = jobListing.children[0];
     jobTitle.innerText = job['jobTitle'];
 
     const jobAddress = jobListing.children[1];
-
     const location = job['jobLocation'];
     jobAddress.innerText = `${location['address']}, ${location['postalCode']}`;
 
+    const jobPay = jobListing.children[2];
+    const pay = job['jobPay'];
+    jobPay.innerText = `${pay['min']} - ${pay['max']} SGD ` +
+      `(${pay['paymentFrequency'].toLowerCase()})`;
+
+    const requirementsList = jobListing.children[3];
+    const fullRequirementsList = getRequirementsList();
+    const requirementsArr = [];
+
+    job['requirements'].forEach((req) => {
+      requirementsArr.push(fullRequirementsList[req]);
+    });
+
+    requirementsList.innerText =
+      `Requirements List: ${requirementsArr.join(', ')}`;
+
     jobListingsElement.appendChild(jobListing);
   });
+
+  jobShowing.innerText = `${jobPageData['range'].minimum} -` +
+    ` ${jobPageData['range'].maximum} ${STRINGS['job-listings-showing']} ` +
+    `${jobPageData['totalCount']}`;
+}
+
+/**
+ * Add the list of jobs that are stored in the database
+ * given the filter fields.
+ */
+async function renderJobListings() {
+  if (!validateFilters()) {
+    return;
+  }
+
+  const sortingParam = document.getElementById('sort-by').value;
+
+  /*
+   * Note that this platform currently only sorts by Salary.
+   * TODO(issue/62): support more filters
+   */
+  if (!sortingParam.includes(SALARY_PARAM)) {
+    console.error('this app currently only sorts by salary');
+    return;
+  }
+
+  const sortByParam = SALARY_PARAM;
+  const orderByParam = sortingParam.substring(7);
+
+  const regionParam = document.getElementById('show-by-region').value;
+  let minLimitParam = document.getElementById('filter-min-limit')
+      .valueAsNumber;
+  let maxLimitParam = document.getElementById('filter-max-limit')
+      .valueAsNumber;
+
+  if (Number.isNaN(minLimitParam)) {
```

Comment:
Since we do this check here instead of inside validateFilters. Does that means that we allow NaN in the limit? Can user enter by mistake '100sgd'? And if we fallback to default values when they're NaN, do we reset those values to the text field?

Reply:
The user should only be able to type numbers into the text field (the input type is number), but even if they type something else, then the minLimit and maxLimit will just reset to the default values.

Reply:
The text fields are just left as blank for the default values. I thought this would be better since it doesn't make sense to explicitly write min=0 and max=214...(whatever the max integer value is for java). Do you think it would be better to have the values filled in?